my-app/src/app/shared/pipes/order-by.pipe.ts

```diff
@@ -0,0 +1,27 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'orderBy'
+})
+export class OrderByPipe implements PipeTransform {
+
+  transform<T>(value: T[], ...args: [keyof T, boolean]): T[] {
+    const [orderBy, asc] = args;
+    let result = value.sort((a: T, b: T) => {
+      if (a[orderBy] > b[orderBy]) {
```

Comment:
as `orderBy` should be a number (maybe you'll be able to add a typing for it ;))  `a[orderBy] - b[orderBy]` should be enough for sorting